addons/event/tests/test_event_flow.py

```diff
@@ -1,18 +1,102 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import datetime
+from datetime import datetime, timedelta
 
 from dateutil.relativedelta import relativedelta
 
+from odoo import Command
 from odoo.addons.event.tests.common import TestEventCommon
 from odoo.exceptions import ValidationError
+from odoo.fields import Datetime
 from odoo.tests.common import Form
 from odoo.tools import mute_logger
 
 
 class TestEventUI(TestEventCommon):
 
+    def test_event_event_type_id_edition(self):
+        """ Ensure that event_mail_ids and event_ticket_ids are correctly computed
+        during the event creation/edition (Form), when switching event_type_id
+        """
+
+        et_ticket = self.env['event.type'].create({
+            'name': 'use_ticket',
+            'use_ticket': True,
+            'use_mail_schedule': False,
+            'auto_confirm': False
+        })
+        et_mail = self.env['event.type'].create({
+            'name': 'use_mail_schedule',
+            'use_ticket': False,
+            'use_mail_schedule': True,
+            'auto_confirm': False
+        })
+        et_none = self.env['event.type'].create({
+            'name': 'use_none',
+            'use_ticket': False,
+            'use_mail_schedule': False,
+            'auto_confirm': True
+        })
+        event = self.env['event.event'].create({
+            'name': 'Event Update Type',
+            'date_begin': Datetime.to_string(datetime.today() + timedelta(days=1)),
+            'date_end': Datetime.to_string(datetime.today() + timedelta(days=15)),
+            'event_type_id': et_none.id
+        })
+        event.write({
+            'event_mail_ids': [
+                Command.clear(),
+                Command.create({
+                    'notification_type': 'mail',
+                    'interval_unit': 'now',
+                    'interval_type': 'after_sub',
+                    'template_id': self.env.ref('event.event_subscription').id,
+                })
+            ],
+            'event_ticket_ids': [
+                Command.clear(),
+                Command.create({
+                    'name': 'saved_ticket',
+                    'seats_max': 30,
+                })
+            ]
+        })
+        mail = event.event_mail_ids.filtered(lambda m: m.notification_type == 'mail')
+        registration = self._create_registrations(event, 1)
+        self.assertEqual(registration.state, 'open')  # event auto confirms
+        ticket = event.event_ticket_ids.filtered(lambda t: t.name == 'saved_ticket')
+        registration.write({'event_ticket_id': ticket.id})
+        event_form = Form(event)
+
+        def assert_base_state():
```

Comment:
Wow ... witness markers ? I don't understand the explanation :D . Not sure to understand what "base state" this method is asserting.